lib/api/dsl/methods.js

```diff
@@ -751,11 +747,93 @@ module.exports = methods = {
 
   /**
    * Return true only if the point in field is in a range from a specific point
-   */
-  geoDistanceRange: function () {
-    var deferred = q.defer();
+  * @param {String} roomId
+   * @param {String} collection
+   * @param {Object} filter given by user on subscribe
+   * @param {Boolean} not if not is true, invert the boolean result
+   * @return {Promise} the formatted filter that need to be added to the room
+  */
+  geoDistanceRange: function (roomId, collection, filter, not) {
+    var
+      curriedFunctionName,
+      deferred = q.defer(),
+      fieldName,
+      formattedFilters = {},
+      geoFilter,
+      point,
+      lat,
+      lon,
+      from,
+      to,
+      result;
+
+    if (_.isEmpty(filter)) {
+      deferred.reject(new BadRequestError('Missing filter'));
+      return deferred.promise;
+    }
+
+    fieldName = Object.keys(filter)[0];
+    if (fieldName === 'from' || fieldName === 'to') {
+      fieldName = Object.keys(filter)[1];
+    }
+    if (fieldName === 'from' || fieldName === 'to') {
+      fieldName = Object.keys(filter)[2];
+    }
 
-    deferred.reject(new KuzzleError('geoDistanceRange is not implemented yet.'));
+    geoFilter = filter[fieldName];
```

Comment:
As it is, ``fieldName`` may be undefined, leading to a ``TypeError`` when initializing ``geoFilter``.

For instance with this filter:

```json
{ "from": "foobar" }
```